framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3499,6 +3513,170 @@ private boolean isListForRepeatedAnnotationImplementation(DeclaredType annotatio
         return result;
     }
 
+    /**
+     * Whether or not the {@code annotatedTypeMirror} has an implicit qualifier parameter.
+     *
+     * @param annotatedTypeMirror AnnotatedTypeMirror to check
+     * @param top the top of the hierarchy to check
+     * @return true if the type has a qualifier parameter
+     */
+    public boolean hasQualifierParameterInHierarchy(
```

Comment:
Many of these methods look like they could be static. Maybe we could move them to a new helper class instead of adding them to AnnotatedTypeFactory?

Reply:
They can't be static; they all use `AnnotatedTypeFactory#isSupportedQualifier(AnnotationMirror)` which is not static.